cpu/sensors.py

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Harish <harisrir@linux.vnet.ibm.com>
+#
+
+"""
+Test for sensors command
+"""
+import platform
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+# TODO: Add possible errors of sensors command
+ERRORS = ['I/O error']
+
+
+class Sensors(Test):
+
+    """
+    Run sensors command on ppc architectures.
+    """
+
+    @staticmethod
+    def check_errors(cmd):
+        """
+        Checks any errors in command result
+        """
+        errs = []
+        s_output = process.system_output(cmd)
+        for error in ERRORS:
+            if error in s_output:
+                errs.append(error)
+        return errs
+
+    def test(self):
+        """
+        Test for sensors command
+        """
+        s_mg = SoftwareManager()
+        d_distro = distro.detect()
+        if d_distro.name == "Ubuntu":
+            if not s_mg.check_installed("lm-sensors") and not s_mg.install(
+                    "lm-sensors"):
+                self.error('Need sensors to run the test')
+        else:
+            if not s_mg.check_installed("lm_sensors") and not s_mg.install(
+                    "lm_sensors"):
+                self.error('Need sensors to run the test')
```

Comment:
The need of the check is not under discussion. I think we all agree it's needed. The way it's checked is the point. Personally, I like this approach:
https://github.com/avocado-framework/avocado-misc-tests/blob/master/generic/gdb.py#L33-L43
It does not mean you have to change yours, since it's working.
And for a even more clean way to do the dependency check/install (in the future), I created this card:
https://trello.com/c/wpOjFiHl/791-make-the-test-dependencies-installation-part-of-softwaremanager